### DISCUSSION TITLE
Trying to get the Chaos Legion workaround to cooperate with my custom mapper settings

### DISCUSSION THREAD

**Discussion by ViperAcidZX** [samuelgr/Xidi#57] [Q&A] [answered]
Carrying over the conversation from the issue thread, for some reason the workaround to get Gamepad 2 configuration not appearing for Chaos Legion prevents my custom mapper bindings from working with the workaround. I have my Xidi configuration in the attachment of this post to see what could be the issue with my configuration that is preventing the workaround for Gamepad 2 not appearing from messing up my bindings. I had initially made this for v4.1.1 of Xidi.

[Xidi_ChaosLegionPC.zip](https://github.com/samuelgr/Xidi/files/11416107/Xidi_ChaosLegionPC.zip)

**Comment by samuelgr** [comment]
For convenience, here is the contents of the config file.

```ini
[Mapper]
Type 			    = ChaosLegion

[CustomMapper:ChaosLegion]
Template		    = ExtendedGamepad

; Jump / Confirm / Skip FMV
ButtonA             = Button(3)

; Lock-on Shot / Cancel
ButtonB             = Button(2)

; Attack
ButtonX             = Button(4)

; Attack Orders / Confirm
ButtonY             = Button(1)

; Mode Change
ButtonLB            = Button(7)

; Parallel Movement
ButtonRB            = Button(8)

; Switch Crests
TriggerLT           = Button(5)

; Force Styles
TriggerRT           = Button(6)

; Map
ButtonBack          = Keyboard(End)

; Pause
ButtonStart         = Button(9)

; Not used
ButtonLS            = Button(11)
ButtonRS            = Button(12)

; Move Player (Digital)
; By default POV hat in the PC version of Chaos Legion moves the camera.
; This change to the POV hat makes it match with the PlayStation 2 version's controls.
DpadUp              = Axis(Y, -)
DpadDown            = Axis(Y, +)
DpadLeft            = Axis(X, -)
DpadRight           = Axis(X, +)

; Move Player
StickLeftX          = Axis(X)
StickLeftY          = Axis(Y)

; Camera Control
StickRightX         = Axis(Z)
StickRightY         = Axis(RotZ)
```

**Comment by samuelgr** [comment, marked as answer]
This game is challenging to get working the way you want because it has implementation quirks. Here is what I've learned so far:
 - "Gamepad 2" is only selectable if the controller has a POV hat. The POV hat maps to camera control.
 - In "Gamepad 2" mode the game will read horizontal camera rotation from the Z axis even if the controller doesn't actually have a Z axis (in this case the camera will just spin endlessly to the right).
 - When using the right stick for camera control, vertical camera rotation is analog but horizontal camera movement is digital, 

What you can try doing is giving the virtual controller a POV hat that does nothing. One way of doing that is to change your "not used" LS button:

```ini
; Not used
; We need to add a POV hat that does nothing so that "Gamepad 2" can be selected.
; Adding both directions at the same time means that there is never any actual POV input.
ButtonLS            = Compound( Button(11), Pov(Left), Pov(Right) )
ButtonRS            = Button(12)
```

Another thing: since you provided an element mapper for every controller component you don't need a template. You can get rid of this:

```ini
Template		    = ExtendedGamepad
```

**Comment by ViperAcidZX** [comment]
I tried the suggestions you made and it helped out, thank you!
